src/Analysis/Ast/Impl/Types/PythonSuperType.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Python.Core;
+
+namespace Microsoft.Python.Analysis.Types {
+    internal sealed class PythonSuperType : PythonType, IPythonSuperType {
+        /// <summary>
+        /// more info at https://docs.python.org/3/library/functions.html#super
+        /// </summary>
+        /// <param name="location"></param>
+        /// <param name="mro">Should be a list of IPythonType</param>
+        public PythonSuperType(Location location, IReadOnlyList<IPythonType> mro)
+            : base("super", location, string.Empty, BuiltinTypeId.Type) {
+            Mro = mro;
+        }
+
+        public override string QualifiedName {
+            get {
+                return $"{this.GetQualifiedName()}[{string.Join(",", Mro.Select(t => t.QualifiedName))}]";
```

Comment:
I don't think you need to qualify the name with a location or anything, since the "super type" isn't defined anywhere except in our code. I was imagining this qualified name being something as simple as `s:[a,bunch,of,things]`, rather than calling `GetQualifiedName`.

Reply:
I'm running into issues using "s" and ```DeconstructQualifiedName```

Reply:
Yeah, that function expects a certain number of elements in the name, as everything has had a declaring module up until now.

Maybe you can just leave the declaring module as it is and still special case it, i.e. create `s:<module>:[MRO]` which you can expand into `PythonSuperType`, and don't allow any part of the code to do a lookup into the module for the value.